x/posts/internal/keeper/keeper.go

```diff
@@ -139,7 +139,20 @@ func (k Keeper) GetPostsFiltered(ctx sdk.Context, params types.QueryPostsParams)
 			matchCreator = params.Creator.Equals(post.Creator)
 		}
 
-		if matchParentID && matchCreationTime && matchAllowsComments && matchSubspace && matchCreator {
+		// match hashtags if provided
+		if len(params.Hashtags) > 0 {
+			postHashtags := post.GetPostHashtags()
+			matchHashtags = len(postHashtags) == len(params.Hashtags)
+			sort.Strings(postHashtags)
+			sort.Strings(params.Hashtags)
+			if matchHashtags {
+				for index, hashtag := range params.Hashtags {
+					matchHashtags = postHashtags[index] == hashtag
+				}
+			}
```

Comment:
We need to keep track of all the iterations, and we can remove the first `if` and the number of hashtags will not be that large IMO: 

```suggestion
			for index, hashtag := range params.Hashtags {
					matchHashtags &= postHashtags[index] == hashtag
			}
```